### PR TITLE
fix: monorepo selfheal should not happens if manifest is the same

### PR DESCRIPTION
**Pull request by pasha-codefresh**

### CODE REVIEW

**Comment by pasha-codefresh** [comment]
@CodiumAI-Agent /review

File: controller/appcontroller.go [hunk 1]
```diff
@@ -1959,8 +1959,11 @@ func alreadyAttemptedSync(app *appv1.Application, commitSHA string, commitSHAsMS
 			return false, ""
 		}
 	} else {
-		if app.Status.OperationState.SyncResult.Revision != commitSHA {
-			return false, ""
+		manifestsChangedMap := app.Status.Sync.ManifestsChanged
+		if manifestsChangedMap[commitSHA] {
+			if app.Status.OperationState.SyncResult.Revision != commitSHA {
```
Comment (oleksandr-codefresh) [inline controller/appcontroller.go:1965]: make fix for multi-sourced also

Comment (pasha-codefresh) [reply controller/appcontroller.go:1964]: I will do it in next PR, when will push to OSS. I am doing it for priceline right now

Comment (pasha-codefresh) [reply controller/appcontroller.go:1965]: i will do it when will open pr to oss, for now it is optional

**Pull request merged**